src/Security/Authorization/Policy/src/AuthorizationAppBuilderExtensions.cs

```diff
@@ -23,7 +25,22 @@ public static IApplicationBuilder UseAuthorization(this IApplicationBuilder app)
                 throw new ArgumentNullException(nameof(app));
             }
 
+            VerifyServicesRegistered(app);
+
             return app.UseMiddleware<AuthorizationMiddleware>();
         }
+
+        private static void VerifyServicesRegistered(IApplicationBuilder app)
+        {
+            // Verify that AddAuthorizationPolicy was called before calling UseAuthorization
+            // We use the AuthorizationPolicyMarkerService to ensure all the services were added.
+            if (app.ApplicationServices.GetService(typeof(AuthorizationPolicyMarkerService)) == null)
```

Comment:
I've seen/used this pattern quite a few times now. @davidfowl Is it worth an `IsRegistered<TService>` extension method?

Reply:
Or maybe a `bool TryGetService<TService>(this IServiceProvider provider, out TService service)` (or both (the former calling the latter))?

Reply:
Nah